### ISSUE TITLE
Refactor GPT code to be able to run different configurations

### ISSUE THREAD

**Issue by varun-suresh** [varun-suresh/ml-practice#9]
1. Fine-tuning
2. Fine-tuning with LoRA
3. Zero Shot inference
4. Fine-tuning just the classification layer

Other things to fix:
1. Train Config and Model config should have nothing in common
2. Optimizer and Scheduler should process from where they left off (i.e when running from a saved checkpoint, the learning rate schedule should be followed)
